Nasty hack by Robert Scheck <robert@fedoraproject.org> for UnrealIRCd >= 6.1.8 which disables all
default upstream GeoIP functionalities in UnrealIRCd, because:

As of writing, the origin of the two GeoIP Legacy databases provided by UnrealIRCd upstream is, at
least to me as packager, unknown. MaxMind retired and removed the (free) GeoLite Legacy databases
already in April 2018. The (free) GeoLite2 databases (to be used with libmaxminddb) are maintained
by MaxMind, but due to legal implications these GeoLite2 databases are no longer available under
the Creative Commons Attribution-ShareAlike 4.0 International License (since December 2019). While
the (free) GeoLite2 databases can be still downloaded without charges, it's necessary to register a
MaxMind account and to obtain a license key. Additionally, an end-user license agreement containing
the relevant data processing provisions that MaxMind and the user/admin (= you) need to comply with
applicable data privacy regulations needs to be signed (https://www.maxmind.com/en/geolite2/eula).
Technically it's indeed possible for UnrealIRCd upstream to keep the GeoIP Legacy databases up-to-
date using other sources themself, however:

UnrealIRCd upstream unfortunately still prefers to use the legacy library GeoIP aka libgeoip (which
reached end-of-life in May 2022; https://github.com/maxmind/geoip-api-c/) rather its successor, the
GeoIP2 library aka libmaxminddb (https://github.com/maxmind/libmaxminddb) by default.

See also:
 - https://www.unrealircd.org/docs/GeoIP (for the different options and the configuration itself)
 - https://blog.maxmind.com/2020/06/01/retirement-of-geoip-legacy-downloadable-databases-in-may-2022/
 - https://blog.maxmind.com/2019/12/18/significant-changes-to-accessing-and-using-geolite2-databases/

--- unrealircd-6.1.8/doc/conf/modules.default.conf		2024-10-11 07:39:38.000000000 +0200
+++ unrealircd-6.1.8/doc/conf/modules.default.conf.geoip	2024-10-12 23:36:25.053387669 +0200
@@ -218,10 +218,10 @@
 loadmodule "extbans/timedban"; /* +b ~time (old ~t mode, timed bans / temporary bans) */
 loadmodule "extbans/partmsg"; /* +b ~partmsg (old ~p mode, hide part/quit message) */
 loadmodule "extbans/securitygroup"; /* +b ~security-group (old ~G mode, security group) */
-loadmodule "extbans/country"; /* +b ~country (country via geoip) */
+// loadmodule "extbans/country"; /* +b ~country (country via geoip) */ /* Disabled by Fedora intentionally, please read the end of this file */
 loadmodule "extbans/msgbypass"; /* +e ~msgbypass (old ~m mode, bypass message restrictions) */
 loadmodule "extbans/flood"; /* +e ~flood (flood exemption for chanmode +f/+F) */
-loadmodule "extbans/asn"; /* +b ~asn (ASN via geoip) */
+// loadmodule "extbans/asn"; /* +b ~asn (ASN via geoip) */ /* Disabled by Fedora intentionally, please read the end of this file */
 loadmodule "extbans/inherit"; /* +b ~inherit (inherit bans from another channel) */
 
 
@@ -288,18 +288,42 @@
 loadmodule "connthrottle"; /* see https://www.unrealircd.org/docs/Connthrottle */
 loadmodule "userip-tag"; /* unrealircd.org/userip tag for ircops (IP address) */
 loadmodule "userhost-tag"; /* unrealircd.org/userhost tag for ircops (real host) */
-loadmodule "geoip-tag"; /* unrealircd.org/geoip tag for ircops (country) */
+// loadmodule "geoip-tag"; /* unrealircd.org/geoip tag for ircops (country) */ /* Disabled by Fedora intentionally, please read the end of this file */
 loadmodule "json-log-tag"; /* unrealircd.org/json-log tag and CAP for ircops */
 loadmodule "issued-by-tag"; /* unrealircd.org/issued-by tag for servers and ircops */
 loadmodule "real-quit-reason"; /* unrealircd.org/real-quit-reason for servers only */
 loadmodule "targetfloodprot"; /* set::anti-flood::target-flood protection */
 loadmodule "watch-backend"; /* used by watch and other modules */
-loadmodule "geoip_base"; /* needed for ALL geoip functions */
+// loadmodule "geoip_base"; /* needed for ALL geoip functions */ /* Disabled by Fedora intentionally, please read the end of this file */
 loadmodule "websocket_common"; /* helper functions for websocket (internal) */
 loadmodule "spamreport"; /* Spam reporting to a blacklist */
 loadmodule "crule"; /* Rules in spamfilter::rule and deny link::rule */
 
-loadmodule "geoip_classic";
+/* As of writing, the origin of the two GeoIP Legacy databases provided by UnrealIRCd upstream is, at
+ * least to me as packager, unknown. MaxMind retired and removed the (free) GeoLite Legacy databases
+ * already in April 2018. The (free) GeoLite2 databases (to be used with libmaxminddb) are maintained
+ * by MaxMind, but due to legal implications these GeoLite2 databases are no longer available under
+ * the Creative Commons Attribution-ShareAlike 4.0 International License (since December 2019). While
+ * the (free) GeoLite2 databases can be still downloaded without charges, it's necessary to register a
+ * MaxMind account and to obtain a license key. Additionally, an end-user license agreement containing
+ * the relevant data processing provisions that MaxMind and the user/admin (= you) need to comply with
+ * applicable data privacy regulations needs to be signed (https://www.maxmind.com/en/geolite2/eula).
+ * Technically it's indeed possible for UnrealIRCd upstream to keep the GeoIP Legacy databases up-to-
+ * date using other sources themself, however:
+ *
+ * UnrealIRCd upstream unfortunately still prefers to use the legacy library GeoIP aka libgeoip (which
+ * reached end-of-life in May 2022; https://github.com/maxmind/geoip-api-c/) rather its successor, the
+ * GeoIP2 library aka libmaxminddb (https://github.com/maxmind/libmaxminddb) by default.
+ *
+ * See also:
+ * - https://www.unrealircd.org/docs/GeoIP (for the different options and the configuration itself)
+ * - https://blog.maxmind.com/2020/06/01/retirement-of-geoip-legacy-downloadable-databases-in-may-2022/
+ * - https://blog.maxmind.com/2019/12/18/significant-changes-to-accessing-and-using-geolite2-databases/
+ */
+
+// loadmodule "geoip_classic"; /* Disabled by Fedora intentionally, requires the (already since May 2022 end-of-life) legacy GeoIP library */
+// loadmodule "geoip_csv"; /* Requires a suitable CSV file and a manual configuration (including a database update mechanism) */
+// loadmodule "geoip_maxmind"; /* Requires unrealircd-maxmind RPM to be installed, GeoLite2 databases and a manual configuration (including a database update mechanism) */
 @if module-loaded("geoip_classic")
 set {
 	geoip-classic {
